### DISCUSSION TITLE
Can 11ty output YAML from a JSON global data source?

### DISCUSSION THREAD

**Discussion by jasonday** [11ty/eleventy#2373] [Q&A]
I've been using a headless CMS with an API as a global data source for my 11ty site. 
I need to switch the site to use YAML files rather than the headless CMS, and I'm wondering if I can have 11ty spit out YAML files since it's already parsing the data sources?

And then I can change my setup to use the YAML files that were output.

**Comment by cfjedimaster** [comment]
11ty can spit out anything, just change your permalink to specify the right filename and write your code to output the YAML content. Make sense?

**Reply by jasonday** [reply]
I assume I'd need something like `js-yaml`?

https://www.11ty.dev/docs/data-custom/#yaml

**Reply by cfjedimaster** [reply]
Maybe I didn't quite understand your question. That library you linked to above would make YAML files available as data. I didn't think you were asking that. I thought you had said, given that you had some set of data, you wanted to output it as YAML. That is what you want to do, right?

**Reply by zachleat** [reply]
Output a template as YAML (using an `.11ty.js` template)

```js
const yaml = require('js-yaml');

exports.data = {
  permalink: "/my-yaml-file.yml"
};
exports.render = (data) {
  // return a yaml.dump of something from data?
}
```

**Reply by jasonday** [reply]
@cfjedimaster correct - I need to output YAML. I linked to the wrong thing, I meant to link to `js-yaml` >  https://www.npmjs.com/package/js-yaml

**Reply by cfjedimaster** [reply]
Then you can do as @zachleat showed, or if you don't want to use a .js template, you can use liquid/etc/etc to dynamically output YAML yourself.
